### PR TITLE
Launch Flutter app in test

### PR DESCRIPTION
**Pull request by stevemessick**
@devoncarew @pq 

Launches a Flutter app in a GUI test.

Includes more deterministic pub sync, via pubspec.yaml editor.

### CODE REVIEW

File: flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package io.flutter.tests.gui
```
Comment (devoncarew) [inline flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt]: nit: copyright

File: flutter-gui-tests/testSrc/io/flutter/tests/gui/fixtures/FlutterMessagesToolWindowFixture.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package io.flutter.tests.gui.fixtures
```
Comment (devoncarew) [inline flutter-gui-tests/testSrc/io/flutter/tests/gui/fixtures/FlutterMessagesToolWindowFixture.kt]: same re: the copyright

File: flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+@RunWithIde(CommunityIde::class)
+class InspectorTest : GuiTestCase() {
+
+  @Test
+  fun importSimpleProject() {
+    // TODO(messick) killall Simulator
+    ProjectCreator.importProject()
+    ideFrame {
+      selectSimulator()
```
Comment (devoncarew) [inline flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt]: It looks like this test is set up to work with the iOS Simulator on macos. We should plan to refactor to instead use the flutter test device (available on all platforms, w/o having to install any iOS or Android tooling).

File: flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+//    val comboBoxActionFixture = ComboBoxActionFixture.findComboBoxByText(robot(), actionToolbarContainer!!, "<no devices>")
+//    comboBoxActionFixture.selectItem(devName)
+    // Need to get focus on the combo box but the ComboBoxActionFixture.click() method is private, so it is inlined here.
+    val selector = button("<no devices>")
+    val comboBoxButtonFixture = JButtonFixture(robot(), selector.target())
+    GuiTestUtilKt.waitUntil("ComboBoxButton will be enabled", Timeouts.minutes02) {
+      GuiTestUtilKt.computeOnEdt { comboBoxButtonFixture.target().isEnabled } ?: false
+    }
+    comboBoxButtonFixture.click()
+    robot().pressAndReleaseKey(KeyEvent.VK_DOWN)
```
Comment (pq) [inline flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt]: I'm guessing this is to select the first entry?

If this ends up being flaky (it sure was w/ SWT), I wonder about just calling the combo box APIs directly.

File: flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    robot().pressAndReleaseKey(KeyEvent.VK_ENTER)
+    robot().waitForIdle()
+  }
+
+  private fun IdeFrameFixture.findActionButtonByActionId(actionId: String): ActionButtonFixture {
+    var button: ActionButtonFixture? = null
+    ideFrame {
+      button = ActionButtonFixture.fixtureByActionId(target(), robot(), actionId)
```
Comment (pq) [inline flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt]: Ah!  Are there other finding methods besides action ID?  (Apropos my comment above.)

File: flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    }
+    return button!!
```
Comment (pq) [inline flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt]: What does using `!!` do here?  Won't there just be a null pointer at the call-site anyway?  (Or is that preferred Kotlin style?)

File: flutter-gui-tests/testSrc/io/flutter/tests/gui/fixtures/FlutterMessagesToolWindowFixture.kt (hunk 2 above)
Comment (stevemessick) [reply flutter-gui-tests/testSrc/io/flutter/tests/gui/fixtures/FlutterMessagesToolWindowFixture.kt]: I will add copyrights.

File: flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt (hunk 3 above)
Comment (stevemessick) [reply flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt]: The equivalent of `flutter -d flutter-tester run`. We'll need to change the plugin to make that work. I'll take a look.

File: flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        return runButton.target().getParent()
+      }
+    })
+    assertNotNull(actionToolbarContainer)
+
+    // These next two lines look like the right way to select the simulator, but it does not work.
+//    val comboBoxActionFixture = ComboBoxActionFixture.findComboBoxByText(robot(), actionToolbarContainer!!, "<no devices>")
+//    comboBoxActionFixture.selectItem(devName)
+    // Need to get focus on the combo box but the ComboBoxActionFixture.click() method is private, so it is inlined here.
+    val selector = button("<no devices>")
```
[the remark this answers is not in the document]
Comment (stevemessick) [reply flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt]: There's a `label-for` property that can be added to labels (JLabel) that connects them to some other field, like text or combo box. Unfortunately, we don't have a label on this combo box. We don't really have much (if any) control over the framework implementation. However, Jetbrains has so far been very responsive to requests I've made.

File: flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt (hunk 4 above)
Comment (stevemessick) [reply flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt]: That might work. I'm hoping the "real" API will be usable soon.

File: flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt (hunk 5 above)
Comment (stevemessick) [reply flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt]: It varies by fixture. Some are more fleshed-out than others. Still early days.

**Pull request merged**